### PR TITLE
Usage examples

### PR DESCRIPTION
**Pull request by jasonaibrahim**
Providing some examples for using the ruby sdk and a README that goes through the example code. This is something I talked about with Tim in an effort to provide more docs and examples for customers.

### CODE REVIEW

File: examples/basic/audio-concat-transcoder.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class AudioConcatTranscoder < MediaTranscoder
+  require 'transloadit'
+  require './media-transcoder'
+
+  # in this example a file is encoded as an mp3, id3 tags are added, and it is stored in s3
+  def transcode!(files)
+    concat = transloadit_client.step('concat', '/audio/concat', {
+      ffmpeg_stack: 'v2.2.3',
```
Comment (rmehner) [inline examples/basic/audio-concat-transcoder.rb]: @tim-kos @kvz you know best, but is this a required param? Or can we leave that out to ease that example a little bit?

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+Finally, we submit the assembly in the same way as the previous example:
+
+```
+assembly = transloadit_client.assembly(steps: [encode_mp3, write_metadata, store])
+assembly.submit! open(file)
+```
+
+### Third example
```
Comment (rmehner) [inline examples/README.md]: Can you please add a new line after a new heading? Makes it easier to read the pure markdown without rendering :) Thanks!

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  use: {
+    steps: files.map.each_with_index do |f, i|
+      { name: ':original', as: "audio_#{i}", fields: "file_#{i}" }
+    end
+  },
+  result: true
+})
+```
+
+Taking a look at the ```concat``` step, we see a different usage of the ```use``` parameter
```
Comment (rmehner) [inline examples/README.md]: Single backticks should be enough here :)

File: examples/basic/audio-concat-transcoder.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class AudioConcatTranscoder < MediaTranscoder
+  require 'transloadit'
+  require './media-transcoder'
```
Comment (rmehner) [inline examples/basic/audio-concat-transcoder.rb]: Do you think `require_relative` would be nicer here?

File: examples/basic/media-transcoder.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class MediaTranscoder
+  def transloadit_client
+    unless @transloadit
+      @transloadit = Transloadit.new({
+        key: ENV.fetch('TRANSLOADIT_KEY'),
+        secret: ENV.fetch('TRANSLOADIT_SECRET')
+      })
+    end
+    @transloadit
```
Comment (rmehner) [inline examples/basic/media-transcoder.rb]: Think the ruby idiom here would be:

```ruby
def transloadit_client
  @transloadit ||= Transloadit.new({
    # ...
  })
end
```

What do you think about that? :)

File: examples/basic/media-transcoder.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  def get_status!(assembly_id)
+    req = Transloadit::Request.new('/assemblies/' + assembly_id.to_s, ENV.fetch('TRANSLOADIT_SECRET'))
+    req.get.extend!(Transloadit::Response::Assembly)
+  end
+end
```
Comment (rmehner) [inline examples/basic/media-transcoder.rb]: Nitpick: if you can, please add newlines to the end of files, thanks!

Depending on your editor, that might be just a setting to have that done automatically.

File: examples/basic/audio-concat-transcoder.rb (hunk 4 above)
Comment (jasonaibrahim) [inline examples/basic/audio-concat-transcoder.rb]: yes, thank you for the tip. updated

File: examples/basic/media-transcoder.rb (hunk 5 above)
Comment (jasonaibrahim) [inline examples/basic/media-transcoder.rb]: 👍  thank you

File: examples/basic/media-transcoder.rb (hunk 6 above)
Comment (jasonaibrahim) [inline examples/basic/media-transcoder.rb]: you got it

**Comment by jasonaibrahim** [comment]
I've made the suggested changes. Let me know what you would like to do with respect to the README.md vs including the text as code comments and I will take care of it

**Comment by rmehner** [comment]
Thanks @jasonaibrahim, two minor things left:

1. Can you please add a [relative link to the examples](https://github.com/blog/1395-relative-links-in-markup-files) in the main README.md?
2. Can you please add the change to the CHANGELOG and give yourself some credit for that?

Will merge once this is done, thank you!

**Comment by jasonaibrahim** [comment]
hi @rmehner ive updated CHANGELOG and README. Let me know if theres anything else!

**Comment by rmehner** [comment]
Top notch, thank you @jasonaibrahim 🍾

**Pull request merged**